ledger/block/src/transaction/merkle.rs

```diff
@@ -191,20 +188,28 @@ impl<N: Network> Transaction<N> {
         let functions = program.functions();
         // Retrieve the verifying keys.
         let verifying_keys = deployment.verifying_keys();
+        // Retrieve the number of functions.
+        let num_functions = functions.len();
 
         // Ensure the number of functions and verifying keys match.
         ensure!(
-            functions.len() == verifying_keys.len(),
-            "Number of functions ('{}') and verifying keys ('{}') do not match",
-            functions.len(),
+            num_functions == verifying_keys.len(),
+            "Number of functions ('{num_functions}') and verifying keys ('{}') do not match",
             verifying_keys.len()
         );
+        // Ensure there are functions.
+        ensure!(num_functions > 0, "Deployment must contain at least one function");
```

Comment:
```suggestion
        ensure!(num_functions != 0, "Deployment must contain at least one function");
```
`clippy` likes to complain about such comparisons, even if it doesn't do so _yet_